docs/docs/reference/named-typeargs-spec.md

```diff
@@ -0,0 +1,56 @@
+---
+layout: doc-page
+title: "Named Type Arguments - More Details"
+---
+
+## Syntax
+
+The addition to the grammar is:
+
+```
+SimpleExpr1       ::=  ...
+                    |  SimpleExpr (TypeArgs | NamedTypeArgs)
+NamedTypeArgs     ::=  ‘[’ NamedTypeArg {‘,’ NamedTypeArg} ‘]’
+NamedTypeArg      ::=  id ‘=’ Type
+```
+
+Note in particular that named arguments cannot be passed to type constructors:
+
+``` scala
+class C[T]
+
+val x: C[T = Int] = // error
+  new C[T = Int] // error
+
+class E extends C[T = Int] // error
+```
+
+## Multiple argument lists
+
+When arguments for some parameters are omitted, it's possible to specify the
+remaining arguments using another argument list:
```

Comment:
I believe this was not intended. In my mind, any missing arguments are inferred, and cannot be specified in a subsequent argument list. Otherwise we'll get an ambiguity with polymorphic apply methods. Example:
```
class C[X, Y] { def apply[Z](x: X, y: Y, z: Z) = (x, y, z) }
def f[X, Y]: C[X, Y] = new C[X, Y]
f[Int, Boolean][String](1, true, "") // OK
f[X = Int](1, true, "") // OK, Y and Z are inferred
f[X = Int][String](1, true, "") // error: String argument is assumed to specify `Y`, not `Z`:
1 |f[X = Int][String](1, true, "")
  |                      ^^^^
  |                      found:    Boolean(true)
  |                      required: String
  |                      
```

Reply:
OK, opened https://github.com/lampepfl/dotty/issues/5328